File: sol/solution/0001-0100/0017/Solution.go
func letterCombinations(digits string) []string {
	if len(digits) == 0 {
		return []string{}
	}
	mapping := map[byte]string{
		'2': "abc",
		'3': "def",
		'4': "ghi",
		'5': "jkl",
		'6': "mno",
		'7': "pqrs",
		'8': "tuv",
		'9': "wxyz",
	}
	var res []string
	var backtrack func(index int, path string)
	backtrack = func(index int, path string) {
		if index == len(digits) {
			res = append(res, path)
			return
		}
		letters := mapping[digits[index]]
		for i := 0; i < len(letters); i++ {
			backtrack(index+1, path+string(letters[i]))
		}
	}
	backtrack(0, "")
	return res
}

File: sol/solution/0001-0100/0017/Solution.cs
public class Solution {
    public IList<string> LetterCombinations(string digits) {
        var result = new List<string>();
        if (string.IsNullOrEmpty(digits)) return result;
        string[] mapping = new string[] {
            "",    // 0
            "",    // 1
            "abc", // 2
            "def", // 3
            "ghi", // 4
            "jkl", // 5
            "mno", // 6
            "pqrs",// 7
            "tuv", // 8
            "wxyz" // 9
        };
        Backtrack(digits, mapping, 0, "", result);
        return result;
    }
    private void Backtrack(string digits, string[] mapping, int index, string current, List<string> result) {
        if (index == digits.Length) {
            result.Add(current);
            return;
        }
        string letters = mapping[digits[index] - '0'];
        foreach (char c in letters) {
            Backtrack(digits, mapping, index + 1, current + c, result);
        }
    }
}